exercises/auth/solution/solution.js

```diff
@@ -1,35 +1,39 @@
 const Hapi = require('hapi');
 const Auth = require('hapi-auth-basic');
 
-var user = { name: 'hapi', password: 'auth' };
-var server = new Hapi.Server();
+const user = { name: 'hapi',  password: 'auth' };
 
-server.connection({
-    host: 'localhost',
-    port: Number(process.argv[2] || 8080)
-});
-
-var validate = (request, username, password, callback) => {
+const validate = async (request, username, password, h) => {
     var isValid = username === user.name && password === user.password;
 
-    return callback(null, isValid, { name: user.name });
+    return { isValid: isValid, credentials: {name: user.name} };
 };
 
-server.register(Auth, (err) => {
-    server.auth.strategy('simple', 'basic', { validateFunc: validate });
-    server.route({
-        method: 'GET',
-        path: '/',
-        config: {
-            auth: 'simple',
-            handler: (request, reply) => {
-                reply();
+(async () => {
```

Comment:
I put some time into thinking about what the entrypoint to a server should look like, and here's what I came-up with: https://github.com/devinivy/boilerplate-api/blob/pal/server/index.js .  The main idea is to add a couple lines to throw unhandled rejections.

This was also a good conversation about it: https://github.com/hapijs/discuss/issues/570